src/parquet/column/scanner.h

```diff
@@ -169,6 +151,31 @@ class TypedScanner : public Scanner {
   inline void FormatValue(void* val, char* buffer, size_t bufsize, size_t width);
 
   T* values_;
+
+  bool NextLevels(int16_t* def_level, int16_t* rep_level) {
+    bool is_required = def_levels_.size() == 0;
+    bool is_repeated = rep_levels_.size() > 0;
+    if (level_offset_ == levels_buffered_) {
+      int16_t* def_levels_data = is_required ? nullptr : &def_levels_[0];
+      int16_t* rep_levels_data = is_repeated ? &rep_levels_[0] : nullptr;
+
+      levels_buffered_ = typed_reader_->ReadBatch(batch_size_, def_levels_data,
+          rep_levels_data, values_, &values_buffered_);
+
+      value_offset_ = 0;
+      level_offset_ = 0;
+      if (!levels_buffered_) {
+        return false;
+      }
+    }
+    *rep_level = is_repeated ?
+        def_levels_[level_offset_] : descr()->max_definition_level();
```

Comment:
`descr()->max_definition_level()` and `descr()->max_repetition_level()` are swapped.